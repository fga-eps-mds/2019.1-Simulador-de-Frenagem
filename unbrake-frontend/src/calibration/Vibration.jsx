import React from "react";
import PropTypes from "prop-types";
import { reduxForm } from "redux-form";
import { withStyles, Grid } from "@material-ui/core";
import { connect } from "react-redux";
import styles from "../components/Styles";
import RealTimeChart from "../components/RealTimeChart";
import { checkbox, field } from "../components/ComponentsForm";
import { changeCalibTest } from "../actions/TestActions";

const invalidId = 0;

export const labelVibration = name => {
  let nameLabel = "";
  switch (name) {
    case "CHVB":
      nameLabel = "Canal de aquisição";
      break;
    case "Vmv":
      nameLabel = "Vibração(mv)";
      break;
    case "PVmv":
      nameLabel = "Plota Vibração(mv)";
      break;
    case "Vg":
      nameLabel = "Vibração(g)";
      break;
    case "PVg":
      nameLabel = "Plota Vibração(g)";
      break;
    case "FCVB":
      nameLabel = "Fator de conversão";
      break;
    case "OFVB":
      nameLabel = "Offset";
      break;
    default:
      nameLabel = "";
      break;
  }
  return nameLabel;
};

const renderField = (states, classes, handleChange) => {
  const type = states;
  type.label = labelVibration(states.name);
  return (
    <Grid alignItems="center" justify="center" container item xs={6}>
      {field(type, classes, handleChange)}
    </Grid>
  );
};

const freeFields = (states, classes, handleChange) => {
  const fields = states.map(value => {
    return (
      <React.Fragment>
        {renderField(value, classes, handleChange)}
      </React.Fragment>
    );
  });
  return fields;
};

const vibrationUnits = (states, classes, handleChange) => {
  const type = states[1];
  type.label = labelVibration(type.name);
  return (
    <Grid alignItems="center" container justify="center" item xs={12}>
      {renderField(states[0], classes, handleChange)}

      <Grid alignItems="center" justify="center" container item xs={6}>
        {checkbox(type, handleChange)}
      </Grid>
    </Grid>
  );
};

const allFields = (states, classes, handleChange) => {
  return (
    <Grid alignItems="center" justify="center" container item xs={12}>
      <Grid item xs={12}>
        {renderField(states[0], classes, handleChange)}
      </Grid>

      {vibrationUnits(states[1], classes, handleChange)}
      {vibrationUnits(states[2], classes, handleChange)}

      {freeFields(states[3], classes, handleChange)}
    </Grid>
  );
};

class Vibration extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      vibration: {
        CHVB: "", // canal de aquisição
        Vmv: "", // Vibration(mv)
        PVmv: false, // plota vibration(mv)
        Vg: "", // vibration(g)
        PVg: false, // plota vibration (g)
        FCVB: "", // fator de conversão
        OFVB: "" // Offset de rotação
      }
    };
    this.handleChange = this.handleChange.bind(this);
  }

  handleChange(event) {
    const { calibId, changeCalib } = this.props;
    const { name, type, value, checked } = event;

    const newValue = type === "checkbox" ? checked : value;
    const vibration = { [name]: newValue };

    this.setState(prevState => ({
      vibration: { ...prevState.vibration, ...vibration }
    }));

    if (calibId > invalidId) changeCalib({ calibId: "" });
  }

  render() {
    const { vibration } = this.state;
    const { CHVB, FCVB, OFVB, Vmv, PVmv, Vg, PVg } = vibration;
    const { classes } = this.props;
    const states = [
      { name: "CHVB", value: CHVB, disable: true },
      [
        { name: "Vmv", value: Vmv, disable: true },
        { name: "PVmv", value: PVmv, disable: false }
      ],
      [
        { name: "Vg", value: Vg, disable: true },
        { name: "PVg", value: PVg, disable: false }
      ],
      [
        { name: "FCVB", value: FCVB, disable: false },
        { name: "OFVB", value: OFVB, disable: false }
      ]
    ];
    return (
      <Grid container xs={12} item justify="center">
        <Grid
          alignItems="center"
          justify="center"
          container
          style={{ marginTop: "10px" }}
        >
          <form className={classes.container}>
            <Grid item xs />
            <Grid container item justify="center" xs={6}>
              {allFields(states, classes, this.handleChange)}
            </Grid>
            <Grid item xs />
          </form>
        </Grid>

        <Grid item container xs={9} justify="center">
          <RealTimeChart />
        </Grid>
      </Grid>
    );
  }
}

const mapDispatchToProps = dispatch => ({
  changeCalib: payload => dispatch(changeCalibTest(payload))
});

const mapStateToProps = state => ({
  calibId: state.testReducer.calibId
});

Vibration.defaultProps = {
  calibId: ""
};

Vibration.propTypes = {
  classes: PropTypes.objectOf(PropTypes.string).isRequired,
  calibId: PropTypes.number,
  changeCalib: PropTypes.func.isRequired
};

const VibrationForm = reduxForm({
  form: "calibration",
  destroyOnUnmount: false
})(Vibration);

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(withStyles(styles)(VibrationForm));
