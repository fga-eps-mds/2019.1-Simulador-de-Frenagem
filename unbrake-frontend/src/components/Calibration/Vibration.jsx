import React from "react";
import PropTypes from "prop-types";
import { reduxForm } from "redux-form";
import { withStyles, Grid } from "@material-ui/core";
import styles from "../Styles";
import RealTimeChart from "../RealTimeChart";
import { checkbox, field } from "./CalibrationComponents";

const renderField = (states, classes, handleChange) => {
  return (
    <Grid alignItems="center" justify="center" container item xs={6}>
      {field(states, classes, handleChange)}

    </Grid>
  );
};

const freeFields = (states, classes, handleChange) => {
  const fields = states.map(value => {
    return (
      <React.Fragment>
        {renderField(value, classes, handleChange)}
      </React.Fragment>
    );
  });
  return fields;
};

const vibrationUnits = (states, classes, handleChange) => {
  return (
    <Grid alignItems="center" container justify="center" item xs={12}>
      {renderField(states[0], classes, handleChange)}

      <Grid alignItems="center" justify="center" container item xs={6}>
        {checkbox(states[1], handleChange)}
      </Grid>
    </Grid>
  );
};

const allFields = (states, classes, handleChange) => {
  return (
    <Grid alignItems="center" justify="center" container item xs={12}>
      <Grid item xs={12}>
        {renderField(states[0], classes, handleChange)}
      </Grid>

      {vibrationUnits(states[1], classes, handleChange)}
      {vibrationUnits(states[2], classes, handleChange)}

      {freeFields(states[3], classes, handleChange)}
    </Grid>
  );
};

class Vibration extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      vibration: {
        CHVB: "", // canal de aquisição
        Vmv: "", // Vibration(mv)
        PVmv: false, // plota vibration(mv)
        Vg: "", // vibration(g)
        PVg: false, // plota vibration (g)
        FCVB: "", // fator de conversão
        OFVB: "" // Offset de rotação
      }
    };
    this.handleChange = this.handleChange.bind(this);
  }

  handleChange(event) {
    const { name, type, value, checked } = event;
    const newValue = type === "checkbox" ? checked : value;
    const vibration = { [name]: newValue };

    this.setState(prevState => ({
      vibration: { ...prevState.vibration, ...vibration }
    }));
  }

  render() {
    const { vibration } = this.state;
    const { CHVB, FCVB, OFVB, Vmv, PVmv, Vg, PVg } = vibration;
    const { classes } = this.props;
    const states = [
      { name: "CHVB", value: CHVB, disable: true },
      [
        { name: "Vmv", value: Vmv, disable: true },
        { name: "PVmv", value: PVmv, disable: false }
      ],
      [
        { name: "Vg", value: Vg, disable: true },
        { name: "PVg", value: PVg, disable: false }
      ],
      [
        { name: "FCVB", value: FCVB, disable: false },
        { name: "OFVB", value: OFVB, disable: false }
      ]
    ];
    return (
      <Grid container xs={12} item justify="center">
        <Grid
          alignItems="center"
          justify="center"
          container
          style={{ minHeight: "60vh" }}
        >
          <form className={classes.container}>
            <Grid item xs />
            <Grid container item justify="center" xs={6}>
              {allFields(states, classes, this.handleChange)}
            </Grid>
            <Grid item xs />
          </form>
        </Grid>

        <Grid item container xs={9} justify="center">
          <RealTimeChart />
        </Grid>
      </Grid>
    );
  }
}

Vibration.propTypes = {
  classes: PropTypes.objectOf(PropTypes.string).isRequired
};

const VibrationForm = reduxForm({
  form: "calibration"
})(Vibration);

export default withStyles(styles)(VibrationForm);
