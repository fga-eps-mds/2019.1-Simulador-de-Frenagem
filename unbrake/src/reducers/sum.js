const sum = (state = {}, action) => { //Jeito de definir uma função
  switch (action.type) {
    case 'ADD':
      console.log(state, action)
      return {...state, sum: +action.n1 + +action.n2};
      
    default:
      return state
  }
}

export default sum 
