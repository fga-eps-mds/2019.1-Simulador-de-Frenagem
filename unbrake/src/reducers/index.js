import { combineReducers } from 'redux'
import sum from './sum'

const Reducer = combineReducers({ //Agrupa os componentes
	sum,
})

export default Reducer;