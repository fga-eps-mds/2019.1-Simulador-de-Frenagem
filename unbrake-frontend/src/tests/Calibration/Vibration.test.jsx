import React from "react";
import { createStore } from "redux";
import { Provider } from "react-redux";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import Vibration from "../../calibration/Vibration";
import ComponentTest from "../ComponentTest";
import reducers from "../../reducer/index";

Enzyme.configure({ adapter: new Adapter() });

describe("<Vibration />", () => {
  const context = React.createContext();
  const store = createStore(reducers);
  const wrapper = shallow(
    <Provider context={context} store={store}>
      <Vibration />
    </Provider>
  );
  ComponentTest(wrapper);
});
