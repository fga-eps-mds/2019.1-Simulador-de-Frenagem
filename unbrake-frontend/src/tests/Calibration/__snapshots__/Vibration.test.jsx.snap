// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Vibration /> render() renders the component 1`] = `
<Hoc
  classes={
    Object {
      "container": "ReduxForm-container-1",
      "dense": "ReduxForm-dense-3",
      "fileInput": "ReduxForm-fileInput-8",
      "file_name": "ReduxForm-file_name-11",
      "grid": "ReduxForm-grid-5",
      "gridButton": "ReduxForm-gridButton-6",
      "input": "ReduxForm-input-9",
      "menu": "ReduxForm-menu-4",
      "rootUploadFile": "ReduxForm-rootUploadFile-10",
      "textField": "ReduxForm-textField-2",
      "title": "ReduxForm-title-7",
    }
  }
  destroyOnUnmount={true}
  enableReinitialize={false}
  forceUnregisterOnUnmount={false}
  form="calibration"
  forwardedRef={null}
  getFormState={[Function]}
  keepDirtyOnReinitialize={false}
  persistentSubmitErrors={false}
  pure={true}
  shouldAsyncValidate={[Function]}
  shouldError={[Function]}
  shouldValidate={[Function]}
  shouldWarn={[Function]}
  submitAsSideEffect={false}
  touchOnBlur={true}
  touchOnChange={false}
  updateUnregisteredFields={false}
/>
`;
